job/strategies/collaborative_filtering.py

```diff
@@ -59,4 +59,4 @@ def generate_embeddings(self):
         model = Trainer(self.train_data, self.experiment_time, _spotlight_transform, self.training_params)
```

Comment:
♻️ Let's reorganize the `strategies` subdirectory so that upon looking at it, we can readily see the three Python modules corresponding to the three strategies in our toolbox. This means moving stuff in `save_defaults.py`, `save_predictions.py` and the `collaborative_filter` directory to its appropriate strategy or strategy-template module (and even deleting them if said module already has a similar method, e.g., `save_predictions`).

Reply:
Just to be sure: should I move class `Trainer` and the functions in `train_model.py` to `collaborative_filtering.py`?

Reply:
Sounds good, yep!